### PR TITLE
create Compressor classes for all built-in algorithms

### PR DESCRIPTION
**Pull request by mattnworb**
- rename NullCompressor to NoopCompressor
- register default compressors for gzip, default, no-op
- use Compressors for zlib compression/decompression

### CODE REVIEW

File: test/core/compression/message_compress_test.cc [hunk 1]
```diff
@@ -312,14 +312,14 @@ static void test_bad_decompression_algorithm(void) {
 }
 
 static void test_compressor_registry(void) {
-  NullCompressor null_compressor;
-  CompressorRegistry::getInstance().register_compressor(&null_compressor);
-
-  GPR_ASSERT((Compressor*)&null_compressor ==
-             CompressorRegistry::getInstance().get_compressor("null"));
+  // test that the built-in compressors are registered
```
Comment (jrussek) [inline test/core/compression/message_compress_test.cc:320]: we should probably also add a test case that expects NULL to make sure it doesn't just always return a random instance. maybe even interrogate the content type from the plugin itself

**Review by jrussek — CHANGES_REQUESTED** [review]

File: test/core/compression/message_compress_test.cc (hunk 1 above)
Comment (mattnworb) [reply test/core/compression/message_compress_test.cc:320]: > we should probably also add a test case that expects NULL to make sure it doesn't just always return a random instance.

good idea, added
